doc/crypto/BN_zero.pod

```diff
@@ -46,10 +48,11 @@ The preferred version of BN_zero() never fails and returns no value.
 
 =head1 BUGS
 
-Someone might change the constant.
+If a B<BIGNUM> is equal to the value of all-bits-set, it will collide
+with the error condition returned by BN_get_word() which uses that
+as an error value.
 
-If a B<BIGNUM> is equal to 0xffffffffL it can be represented as an
-unsigned long but this value is also returned on error.
+B<BN_ULONG> should probably be a typedef.
```

Comment:
I think the `should probably be a typedef` comment should go.  By now some feature tests likely exist that test for `ifdef BN_ULONG`, so it is too late to make this a type instead.  Also, when we add new types, applications that support older releases can test for macros more easily than for new types (instead they'd need to test against the OPENSSL_VERSION_NUMBER that adds the new type).

So all in all while may at some point introduce new types as non-macros, this one should probably stay as-is long-term.